subprojects/packagefiles/libnpupnp/meson.build

```diff
@@ -0,0 +1,215 @@
+project('npupnp', 'cpp',
+  version : '4.1.5',
+  default_options : ['warning_level=1'])
+
+cpp = meson.get_compiler('cpp')
+
+winsock_dep = cpp.find_library('ws2_32', required : host_machine.system() == 'windows')
+iphlpapi_dep = cpp.find_library('iphlpapi', required : host_machine.system() == 'windows')
+nsl_dep = cpp.find_library('nsl', required : host_machine.system() == 'sunos')
+socket_dep = cpp.find_library('socket', required : host_machine.system() == 'sunos')
+
+threads_dep = dependency('threads')
+libmicrohttpd_dep = dependency('libmicrohttpd')
+libcurl_dep = dependency('libcurl')
+expat_dep = dependency('expat', required : get_option('expat'))
+
+if (host_machine.system() == 'windows')
+  add_project_arguments('-DDLL_EXPORT', language : 'cpp')
+endif
+
+add_project_arguments('-DSMALLUT_NO_REGEX', language : 'cpp')
+add_project_arguments('-DSMALLUT_EXTERNAL_INSTANTIATIONS', language : 'cpp')
+
+libnpupnp_incdir = include_directories([ '.', 'inc', 'src/inc' ])
+
+npupnp_sources = [
+  'src/api/upnpapi.cpp',
+  'src/dispatcher/miniserver.cpp',
+  'src/threadutil/ThreadPool.cpp',
+  'src/threadutil/TimerThread.cpp',
+  'src/utils/description.cpp',
+  'src/utils/genut.cpp',
+  'src/utils/httputils.cpp',
+  'src/utils/inet_pton.cpp',
+  'src/utils/md5.cpp',
+  'src/utils/netif.cpp',
+  'src/utils/smallut.cpp',
+  'src/utils/statcodes.cpp',
+  'src/utils/uri.cpp',
+  'src/utils/utf8iter.cpp',
+  'src/webserver/webserver.cpp',
+]
+
+if get_option('debug')
+  npupnp_sources += [
+    'src/api/upnpdebug.cpp',
+  ]
+endif
+
+if get_option('gena')
+  npupnp_sources += [
+    'src/gena/client_table.cpp',
+    'src/gena/gena_callback2.cpp',
+    'src/gena/gena_ctrlpt.cpp',
+    'src/gena/gena_device.cpp',
+    'src/gena/gena_sids.cpp',
+    'src/gena/service_table.cpp',
+  ]
+endif
+
+if get_option('soap')
+  npupnp_sources += [
+    'src/soap/soap_ctrlpt.cpp',
+    'src/soap/soap_device.cpp',
+  ]
+endif
+
+if get_option('ssdp')
+  npupnp_sources += [
+    'src/ssdp/ssdp_ctrlpt.cpp',
+    'src/ssdp/ssdp_device.cpp',
+    'src/ssdp/ssdpparser.cpp',
+    'src/ssdp/ssdp_server.cpp',
+  ]
+endif
+
+if get_option('tools')
+  npupnp_sources += [
+    'src/api/upnptools.cpp',
+  ]
+endif
+
+libnpupnp = library('npupnp',
+  npupnp_sources,
+  include_directories : libnpupnp_incdir,
+  dependencies : [ winsock_dep, iphlpapi_dep, nsl_dep, socket_dep, threads_dep, libmicrohttpd_dep, libcurl_dep, expat_dep ],
```

Comment:
This also has a library version upstream which should be retained, `-version-info 6:1:2` in libtool terms is  `version: '4.2.1'` in meson terms (and produces `libnpupnp.so.4.2.1`).